paddle/majel/allocation.h

```diff
@@ -0,0 +1,39 @@
+#pragma once
+#include <memory>
+
+#include <majel/place.h>
+#include "hl_gpu.h"
```

Comment:
It seems that here we depend on paddle/paddle/cuda, but I don't see anything in CMakeLists.txt specifying this dependency?